src/lib/components/Banner/index.js

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class Banner extends HTMLElement {
```

Comment:
That's essentially right, but yes the nicer approach would be to move d.c.c's implementation to webdev-infra and share it from there. 🙂 

The missing bit in this version is also the user preference, meaning if the banner is clicked away once it should not reappear. That's handled here for d.c.c: https://github.com/GoogleChrome/developer.chrome.com/blob/42bcc49531b53c046d56c05a65971a81ec92ee2a/site/_includes/partials/script.js#L70-L83